File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_WinPropertyArray.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_WinPropertyArray
*/
public class t_WinPropertyArray : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint[]	m_aValue;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_WinPropertyArray() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_aValue = new uint[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			{
				int n = (int) cVariable.getU8();
				if (n > 255) {
					cVariable.error("array size error in m_aValue");
					return false;
				}
				m_aValue = new uint [n];
				for(int i = 0;i < n;++i) {
					m_aValue[i] = cVariable.getU32();
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		{	
				if (m_aValue.Length > 255) {
					cVariable.error("array size error in m_aValue");
					return false;
				}
			int n = m_aValue.Length;
			cVariable.put((byte) n);
			for(int i = 0;i < n;++i) {
				cVariable.put(m_aValue[i]);

			}
		}
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_WinPropertyArray.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_WinPropertyArray
#/
class t_WinPropertyArray:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_aValue = []	#U32
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_aValue = []

	def read(self,cVariable):
		n = cVariable.getU8()
		if (n > 255) :
			cVariable.error('array size error in m_aValue')
			return False
		self.m_aValue = [0]*n
		for i in range(n):
			self.m_aValue[i] = cVariable.getU32()
		return True

	def write(self,cVariable):
		if (len(self.m_aValue) > 255) :
			cVariable.error('array size error in m_aValue')
			return False
		n = len(self.m_aValue)
		cVariable.putU8(n)
		for i in range(n):
			cVariable.putU32(self.m_aValue[i])
		return True

